code/game/objects/items/robot/robot_items.dm

```diff
@@ -34,3 +35,190 @@
+								"<span class='danger'>You shock [M] to no effect.</span>")
+					playsound(loc, 'sound/effects/sparks2.ogg', 50, 1, -1)
+					user.cell.charge -= 500
+					scooldown = TRUE
+					spawn(20)
+						scooldown = FALSE
+		if(3)
+			if(!ccooldown)
+				if(M.health >= 0)
+					if(ishuman(M))
+						user.visible_message("<span class='userdanger'>[user] crushes [M] in their grip!</span>", \
+							"<span class='danger'>You crush [M] in your grip!</span>")
+					else
+						user.visible_message("<span class='userdanger'>[user] crushes [M]!</span>", \
+								"<span class='danger'>You crush [M]!</span>")
+					playsound(loc, 'sound/weapons/smash.ogg', 50, 1, -1)
+					M.adjustBruteLoss(15)
+					user.cell.charge -= 300
+					ccooldown = TRUE
+					spawn(10)
+						ccooldown = FALSE
+
+/obj/item/borg/cyborghug/peacekeeper
+	shockallowed = TRUE
+
+/obj/item/device/harmalarm
+	name = "Sonic Harm Prevention Tool"
+	desc = "Releases a harmless blast that confuses most organics. For when the harm is JUST TOO MUCH"
+	icon_state = "megaphone"
+	var/cooldown = 0
+	var/emagged = FALSE
+
+/obj/item/device/harmalarm/emag_act(mob/user)
+	emagged = !emagged
+	if(emagged)
+		to_chat(user, "<font color='red'>You short out the safeties on the [src]!</font>")
+	else
+		to_chat(user, "<font color='red'>You reset the safeties on the [src]!</font>")
+
+/obj/item/device/harmalarm/attack_self(mob/user)
+	var/safety = !emagged
+	if(cooldown > world.time)
+		to_chat(user, "<font color='red'>The device is still recharging!</font>")
+		return
+
+	if(isrobot(user))
+		var/mob/living/silicon/robot/R = user
+		if(R.cell.charge < 1200)
+			to_chat(user, "<font color='red'>You don't have enough charge to do this!</font>")
+			return
+		R.cell.charge -= 1000
+		if(R.emagged)
+			safety = FALSE
+
+	if(safety == TRUE)
+		user.visible_message("<font color='red' size='2'>[user] blares out a near-deafening siren from its speakers!</font>", \
+			"<span class='userdanger'>The siren pierces your hearing and confuses you!</span>", \
+			"<span class='danger'>The siren pierces your hearing!</span>")
+		for(var/mob/living/carbon/M in get_mobs_in_view(9, user))
+			if(!M.check_ear_prot())
+				M.AdjustConfused(6)
+		audible_message("<font color='red' size='7'>HUMAN HARM</font>")
+		playsound(get_turf(src), 'sound/AI/harmalarm.ogg', 70, 3)
+		cooldown = world.time + 200
+		log_game("[user.ckey]([user]) used a Cyborg Harm Alarm in ([user.x],[user.y],[user.z])")
+		if(isrobot(user))
+			var/mob/living/silicon/robot/R = user
+			if(R.connected_ai)
+				to_chat(R.connected_ai, "<br><span class='notice'>NOTICE - Peacekeeping 'HARM ALARM' used by: [user]</span><br>")
+
+		return
+
+	if(safety == FALSE)
+		user.audible_message("<font color='red' size='7'>BZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZT</font>")
+		for(var/mob/living/carbon/human/H in get_mobs_in_view(9, user))
+			if(istype(H.l_ear, /obj/item/clothing/ears/earmuffs) || istype(H.r_ear, /obj/item/clothing/ears/earmuffs) || H.ear_deaf)

```

Comment:
Isn't this all covered in `check_ear_prot`? If not, why isn't it?

Reply:
The intention of this is to provide full immunity to the emagged siren *only* if you're wearing earmuffs (or deaf); if you have hearing protection (things with HEADBANGPROTECT OR EARBANGPROTECT), then the siren just confuses you; otherwise, it's a full out stun.

Why doesn't it? Probably because it's the way it was made when you added it: https://github.com/ParadiseSS13/Paradise/commit/0c9a469f2cd467f3eab99742e98553693de6a378#diff-df7fe1c87ccd86bb2878257cd89bc8f5R806

Hearing protection has gone through a number of iterations on TG--the most recent one is probably more in line with what you're imagining, but requires a decent refactor of how it's checked for (based on internal organ hearing), so I opted *not* to include that in this PR.